app/src/test/java/lt/vilnius/tvarkau/fragments/ProblemDetailFragmentTest.kt

```diff
@@ -0,0 +1,59 @@
+package lt.vilnius.tvarkau.fragments
+
+import com.nhaarman.mockito_kotlin.any
+import com.nhaarman.mockito_kotlin.verify
+import com.nhaarman.mockito_kotlin.whenever
+import kotlinx.android.synthetic.main.no_internet.*
+import kotlinx.android.synthetic.main.problem_detail.*
+import kotlinx.android.synthetic.main.server_not_responding.*
+import lt.vilnius.tvarkau.backend.ApiResponse
+import lt.vilnius.tvarkau.backend.LegacyApiService
+import lt.vilnius.tvarkau.base.BaseRobolectricTest
+import lt.vilnius.tvarkau.entity.Problem
+import org.assertj.android.api.Assertions.assertThat
+import org.junit.Test
+import rx.Observable
+import javax.inject.Inject
+
+/**
+ * @author Martynas Jurkus
+ */
+class ProblemDetailFragmentTest : BaseRobolectricTest() {
+
+    @Inject
+    lateinit var api: LegacyApiService
+
+    private val Problem.wrapInResponse: Observable<ApiResponse<Problem>>
+        get() {
+            val response = ApiResponse<Problem>()
+            response.result = this
+            return Observable.just(response)
+        }
+
+    override fun setUp() {
+        super.setUp()
+        activity.getTestComponent().inject(this)
+    }
+
+    @Test
+    fun problemWithoutAnswer_displayCorrectLayout() {
+        val problem = Problem()
+        whenever(api.getProblem(any())).thenReturn(problem.wrapInResponse)
```

Comment:
I've tried using resources for actual responses in previous projects, but it's areal hassle to keep them up to date.
It's much easier just to create entities manually as needed.
Otherwise you must have heaps of files with different amounts of data.
For example, just with problem, with problem and answer, different problem types, failure responses, validation responses and so on.